pkg/wallet/account/account.go

```diff
@@ -0,0 +1,265 @@
+	"crypto/rand"
+	"fmt"
+
+	"github.com/awnumar/memguard"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/massalabs/station-massa-wallet/pkg/crypto"
+	"github.com/massalabs/station-massa-wallet/pkg/types"
+	"github.com/massalabs/station-massa-wallet/pkg/types/object"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	AccountLastVersion = 1
+)
+
+type Account struct {
+	Version      *uint8                    `yaml:"Version"`
+	Nickname     string                    `yaml:"Nickname,omitempty"`
+	Address      types.Address             `yaml:"Address,omitempty"`
+	Salt         [16]byte                  `yaml:"Salt,flow"`
+	Nonce        [12]byte                  `yaml:"Nonce,flow"`
+	CipheredData types.EncryptedPrivateKey `yaml:"CipheredData,flow"`
+	PublicKey    types.PublicKey           `yaml:"PublicKey,flow"`
+}
+
+func New(version *uint8, nickname string, address types.Address, salt [16]byte, nonce [12]byte, cipheredData types.EncryptedPrivateKey, publicKey types.PublicKey) (*Account, error) {
+	if version == nil {
+		version = new(uint8)
+		*version = AccountLastVersion
+	}
+
+	if *version > AccountLastVersion {
+		return nil, fmt.Errorf("unsupported account version: %d", *version)
+	}
+
+	if !NicknameIsValid(nickname) {
+		return nil, fmt.Errorf("invalid nickname: %s", nickname)
+	}
+
+	if err := address.Validate(address.Version, object.UserAddress, object.SmartContractAddress); err != nil {
+		return nil, fmt.Errorf("validating address: %w", err)
+	}
+
+	if err := cipheredData.Validate(cipheredData.Version, object.EncryptedPrivateKey); err != nil {
+		return nil, fmt.Errorf("validating encrypted private key: %w", err)
+	}
+
+	if err := publicKey.Validate(publicKey.Version, object.PublicKey); err != nil {
+		fmt.Println("52", publicKey.Version)
+		fmt.Println("53", publicKey.Kind)
+		return nil, fmt.Errorf("validating public key: %w", err)
+	}
+
+	return &Account{
+		Version:      version,
+		Nickname:     nickname,
+		Address:      address,
+		Salt:         salt,
+		Nonce:        nonce,
+		CipheredData: cipheredData,
+		PublicKey:    publicKey,
+	}, nil
+}
+
+func NewEmpty() Account {
+	return Account{
+		Version:  new(uint8),
+		Nickname: "",
+		Address: types.Address{
+			Object: &object.Object{
+				Kind:    object.UserAddress,
+				Version: 0x00,
+				Data:    nil,
+			},
+		},
+		Salt:  [16]byte{},
+		Nonce: [12]byte{},
+		CipheredData: types.EncryptedPrivateKey{
+			Object: &object.Object{
+				Kind:    object.EncryptedPrivateKey,
+				Version: 0x00,
+				Data:    nil,
+			},
+		},
+		PublicKey: types.PublicKey{
+			Object: &object.Object{
+				Kind:    object.PublicKey,
+				Version: 0x00,
+				Data:    nil,
+			},
+		},
+	}
+}
+
+// NewGenerated generates a new account with a random private key. It destroys the guarded password.
+func NewGenerated(guardedPassword *memguard.LockedBuffer, nickname string) (*Account, error) {
+	version := uint8(AccountLastVersion)
+
+	publicKeyBytes, privateKeyBytes, err := ed25519.GenerateKey(nil)
```

Comment:
You should add a security comment explaining that the entropy is read from [crypto/rand.Reader](https://pkg.go.dev/crypto/rand#Reader) or even better, diretcly use rand as you already import it for the salt and the nonce